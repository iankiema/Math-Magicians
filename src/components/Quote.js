import React, { useState, useEffect } from 'react';

const Quote = () => {
  const [quote, setQuote] = useState('');
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState(null);

  const fetchRandomQuote = async () => {
    try {
      const apiUrl = 'https://api.api-ninjas.com/v1/quotes?category=success';
      const apiKey = 'IWrsocgunOOPSPX1lY4wpg==tgIScuaOsYNEJMj8';
      const response = await fetch(apiUrl, {
        headers: {
          'X-Api-Key': apiKey,
        },
      });

      if (!response.ok) {
        throw new Error(`Error: ${response.status} ${await response.text()}`);
      }

      const responseData = await response.json();
      setQuote(responseData[0].quote || 'No joke available');
      setIsLoading(false);
    } catch (err) {
      setError(err);
      setIsLoading(false);
    }
  };

  useEffect(() => {
    fetchRandomQuote();
  }, []);

  if (isLoading) {
    return <div className="quote">Loading...</div>;
  }

  if (error) {
    return (
      <div className="quote">
        Error:
        {' '}
        {error.message}
      </div>
    );
  }

  return (
    <div className="quote">
      <blockquote>{quote}</blockquote>
    </div>
  );
};

export default Quote;
