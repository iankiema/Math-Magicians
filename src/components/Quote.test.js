import React from 'react';
import { render } from '@testing-library/react';
import Quote from './Quote';

// Mock the fetch function globally
global.fetch = jest.fn(() => Promise.resolve({
  ok: true,
  json: () => Promise.resolve([{ quote: 'Fake quote for testing' }]),
}));

describe('Quote component', () => {
  it('should match snapshot', async () => {
    const { asFragment } = render(<Quote />);

    // Use toMatchSnapshot to compare the rendered output with a snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
