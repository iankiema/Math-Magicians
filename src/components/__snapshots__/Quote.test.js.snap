// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="quote"
  >
    Loading...
  </div>
</DocumentFragment>
`;
